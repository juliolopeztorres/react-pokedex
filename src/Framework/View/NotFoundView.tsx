import React from 'react'
import { hot } from "react-hot-loader/root";
import { Link } from "react-router-dom";
import { getRoute } from "../Service/RouteService";

const NotFoundView = () => <div className='p-4'>
  <span className='text-xl'><span className="text-orange-600">404 -</span> Route not found</span>
  <div className='text-gray-600 dark:text-gray-200 text-sm mt-2'>
    Don't know where to go?
    <Link to={getRoute('home')} className="
    ml-2 skew-y-3 inline-block p-1
    bg-orange-100 hover:bg-orange-200
    text-orange-600
    dark:bg-orange-700 dark:hover:bg-orange-600 dark:text-orange-200
    " style={{cursor: 'pointer'}}>
      <div className="-skew-y-3">Take me home</div>
    </Link>
  </div>
</div>

export default hot(NotFoundView)
