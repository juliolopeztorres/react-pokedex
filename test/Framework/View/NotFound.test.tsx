import React from 'react'
import { BrowserRouter } from "react-router-dom"
import { createRoot, Root } from "react-dom/client";
import { act } from "@testing-library/react";

import NotFoundView from "../../../src/Framework/View/NotFound";

let root: Root
let container: HTMLElement
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);

  root = createRoot(container)
});

afterEach(() => {
  // cleanup on exiting
  act(() => root.unmount())

  container.remove();
});

it('should render', () => {
  act(() => root.render(
      <BrowserRouter>
        <NotFoundView/>
      </BrowserRouter>,
    )
  )

  expect(container.textContent).toContain('404 - Route not found')
  expect(container.textContent).toContain('Don\'t know where to go?')

  const a = container.querySelector('a')
  expect(a).not.toBeNull()

  expect(a!.textContent).toContain('Take me home')
  expect(a!.hasAttribute('href')).toBe(true)
  expect(a!.getAttribute('href')).toBe('/')

  expect(container).toMatchSnapshot()
})
