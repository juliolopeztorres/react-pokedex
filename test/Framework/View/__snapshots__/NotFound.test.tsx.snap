// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="p-4"
  >
    <span
      class="text-xl"
    >
      <span
        class="text-orange-600"
      >
        404 -
      </span>
       Route not found
    </span>
    <div
      class="text-gray-600 dark:text-gray-200 text-sm mt-2"
    >
      Don't know where to go?
      <a
        class="
    ml-2 skew-y-3 inline-block p-1
    bg-orange-100 hover:bg-orange-200
    text-orange-600
    dark:bg-orange-700 dark:hover:bg-orange-600 dark:text-orange-200
    "
        href="/"
        style="cursor: pointer;"
      >
        <div
          class="-skew-y-3"
        >
          Take me home
        </div>
      </a>
    </div>
  </div>
</div>
`
